### ISSUE TITLE
Parse error at "allow-cmd true"

### ISSUE THREAD

**Issue by MaxGyver83** [david-janssen/kmonad#74]
Hello David,

thanks for this project! I have always thought that defining keyboard layouts in xkb is too complicated and many features are missing! So I'm curious how well it works with kmonad.

I have

* downloaded the binary for Linux and made it executable,
* added myself to the `input` group and added a udev rule
* cloned the project,
* copied the tutorial into `~/.config/kmonad/`,
* set my input device file,
* replaced `/run/current-system/sw/bin/` with `/bin/` (was it meant like that?).

But I get this error:

```sh
~ ▶ kmonad-0.4.0-linux ~/.config/kmonad/tutorial.kbd
kmonad-0.4.0-linux: Parse error at 151:3:
    |
151 |   allow-cmd true
    |   ^
unexpected 'a'
expecting "cmp-seq", "fallthrough", "init", "input", "output", or ')'
```

This is how my `defcfg` looks like:

```lisp
(defcfg
  ;; For Linux
  ;; input  (device-file "/dev/input/by-id/usb-04d9_daskeyboard-event-kbd")
  input  (device-file "/dev/input/by-path/platform-i8042-serio-0-event-kbd")

  output (uinput-sink "My KMonad output"
    ;; To understand the importance of the following line, see the section on
    ;; Compose-key sequences at the near-bottom of this file.
    ;; "/run/current-system/sw/bin/sleep 1 && /run/current-system/sw/bin/setxkbmap -option compose:ralt")
    "/bin/sleep 1 && /bin/setxkbmap -option compose:ralt")

  ;; For Windows
  ;; input  (low-level-hook)
  ;; output (send-event-sink)

  ;; For MacOS
  ;; input  (iokit-name "my-keyboard-product-string")
  ;; output (kext)

  ;; Comment this is you want unhandled events not to be emitted
  fallthrough true

  ;; Set this to false to disable any command-execution in KMonad
  allow-cmd true
)
```

When I comment out `allow-cmd true`, I get this error instead:

```
kmonad-0.4.0-linux: Parse error at 827:8:
    |
827 |   dat (cmd-button "date >> /tmp/kmonad_example.txt")   ;; Append date to tmpfile
    |        ^
Unrecognized compose-char
```

**Comment by david-janssen** [comment]
Oops, my bad.

The command-button feature was added very recently, and is not contained in the precompiled binaries. So, if you don't care about running shell commands you can either edit the config to not include those, (or even easier, checkout the config with the `0.4.0` tag). If you really do want command buttons, you can either compile `master` yourself, or wait a few days for me to upload `0.4.1`, which I'll do either tomorrow or Monday, and use those binaries instead.

In the future I should really leave master as a reflection of the latest binary release, and leave all new features on develop until I am ready to recompile and release.

Sorry for the confusion.

David

**Comment by slotThe** [comment]
On Sat, Sep 12 2020 04:45, David Janssen wrote:
> In the future I should really leave master as a reflection of the
> latest binary release, and leave all new features on develop until I
> am ready to recompile and release.

I think the better option (imo anyways, feel free to disregard this)
would be a proper changelog with all the non-trivial feature changes
since the last release.  This way master wouldn't just be "this is what
we used to be" (this job is, I think, already done quite well by tagged
releases) and there would be one place to go in order to see what
changed (as opposed to reading through pages of commits on develop).

**Comment by david-janssen** [comment]
Noted. Other people mentioned this as well, so I'll be adding a changelog soon.

**Comment by david-janssen** [comment]
@MaxGyver83: Hey Max, so I did just upload a new `0.4.1` release that should work with the most recent config-file under `master`. Give it a shot and let me know if you have any further issues.

@slotThe: So, I added the CHANGELOG file, it is still quite sparse, but we'll populate it from now on. I will still be sticking to the 'changes go in develop, and then we release stuff onto master', just for ease of use for people (i.e. all the documentation on the front of the Github interface lines up with the most recent binary). Anyone who will want to use more advanced features will have to compile themselves anyways, so they should be savvy enough to checkout `develop`. But the idea of adding a changelog was spot on, much appreciated.
